pandas/tests/indexes/multi/test_setops.py

```diff
@@ -20,21 +20,17 @@ def test_set_ops_error_cases(idx, case, sort, method):
 
 @pytest.mark.parametrize("sort", [None, False])
 def test_intersection_base(idx, sort):
-    first = idx[:5]
-    second = idx[:3]
-    intersect = first.intersection(second, sort=sort)
-
-    if sort is None:
-        tm.assert_index_equal(intersect, second.sort_values())
-    assert tm.equalContents(intersect, second)
+    first = idx[2::-1]  # first 3 elements reversed
+    second = idx[:5]
 
-    # GH 10149
-    cases = [klass(second.values) for klass in [np.array, Series, list]]
-    for case in cases:
-        result = first.intersection(case, sort=sort)
+    array_like_cases = [klass(second.values) for klass in [np.array, Series, list]]
```

Comment:
can you parameterize this test (e.g. klass)

Reply:
done.

The tests for [difference](https://github.com/jeffzi/pandas/blob/83518d1f33cb6f443509bad75c712e28618b5038/pandas/tests/indexes/multi/test_setops.py#L76) and [symmetric_difference](https://github.com/jeffzi/pandas/blob/83518d1f33cb6f443509bad75c712e28618b5038/pandas/tests/indexes/multi/test_setops.py#L99) use a similar loop. I could also refactor them to parametrize if you want the tests to be consistent.